### ISSUE TITLE
Load balancer stud restart problem

### ISSUE THREAD

**Issue by toros11** [axsh/wakame-vdc#523]
Stud frequently fails to restart with upstart due to the way stud handles and child processes. It creates a child process of itself, which upstart does not know about which results in the restart process being attempted before all of the processes have finished terminating.

A possible workaround is to set the respawn limit to unlimited as with the default value the child process will not yet be shutdown completely when the limit is reached.

For the future, looking at how the shutdown process is being handled and a solution which blocks the restart process until both child and parent process of stud are terminated should be considered.

**Comment by Metallion** [comment]
Another thing I want to try is to update the load balancer image to Centos 7 and use systemd instead of upstart. Of course chances are the issue will persist since it's caused by stud not waiting on its child processes properly.

**Comment by hansode** [comment]
When we developed load-balancer feature, the haproxy was `1.4`.
We can use haproxy `1.5` which supports ssl/https.
So we can make new loadbalance appliance with haproxy `1.5` only without stud.

**Comment by hansode** [comment]
Current lb demo image `wmi-haproxy1d64` already uses haroxy `1.5`.

**Comment by Metallion** [comment]
So that gives us the following todo list:

* Reconfigure the load balancer image to use haproxy for ssl instead of stud.
* Remove stud.
* Update the lb image to centos 7 while we're at it.

**Comment by Metallion** [comment]
This has been fixed. I will write a new issue for the STUD replacement.
